Src/CsvLINQPadDriver/SchemaBuilder.cs

```diff
@@ -75,28 +71,21 @@ internal static List<ExplorerItem> GetSchemaAndBuildAssembly(ICsvDataContextDriv
         /// <param name="code"></param>
         /// <param name="name"></param>
         /// <returns></returns>
-        private static string[] BuildAssembly(string code, AssemblyName name) 
+        private static string[] BuildAssembly(string code, AssemblyName name)
         {
-            CompilerResults results;
-            using (var codeProvider = new CSharpCodeProvider(new Dictionary<string, string>() { { "CompilerVersion", "v4.0" } })) {
-                var options = new CompilerParameters(new string[]
-                    {
-                        typeof(SchemaBuilder).Assembly.Location,
-                        typeof(CsvHelper.CsvReader).Assembly.Location,
-                        "System.dll", "System.Core.dll", "System.Xml.dll", "System.Data.Services.Client.dll",
-                    }){
-                    IncludeDebugInformation = true, 
-                    OutputAssembly = name.CodeBase,
-                    CompilerOptions = @"/doc:""" + Path.ChangeExtension( name.CodeBase, "xml") + @"""",
-                };
-                results = codeProvider.CompileAssemblyFromSource(options, code);
-            }
-
-            if (!results.Errors.HasErrors)
-                return null;
-
-            return results.Errors.OfType<CompilerError>().Where(e => !e.IsWarning).Select(e => string.Format("{0},{1}: {2}", e.Line, e.Column, e.ErrorText)).ToArray();
+            var referencedAssemblies = DataContextDriver.GetCoreFxReferenceAssemblies().Concat(new []
+            {
+                typeof(SchemaBuilder).Assembly.Location,
+                typeof(CsvHelper.CsvReader).Assembly.Location
+            }).ToArray();
+            var result = DataContextDriver.CompileSource(new CompilationInput()
```

Comment:
We could probably use Roslyn explicitly, but linqpad DataContextDriver helper provides this functionality as well